src/Text/Markdown/SlamDown/Parser.purs

```diff
@@ -282,11 +289,35 @@ parseBlocks (Cons (CLinkReference b) cs) =
 parseBlocks (Cons _ cs) =
   parseBlocks cs
 
+{-
+
+data Block
+  = Paragraph (List Inline)
+  | Header Int (List Inline)
+  | Blockquote (List Block)
+  | Lst ListType (List (List Block))
+  | CodeBlock CodeBlockType (List String)
+  | LinkReference String String
+  | Rule
+-}
+
+validateBlock :: Block -> V.V (Array String) Block
```

Comment:
This is not a big deal since our Markdown snippets are not large, but PureScript's `Array` has `O(n)` performance characteristics for append. Until then, `List` can of course build things backward in `O(1)`, and [purescript-catenable](https://github.com/ethul/purescript-catenable), when finished, promises `O(1)` snocs / cons, but it's not ready yet.